pkg/build/api/validation/validation_test.go

```diff
@@ -67,7 +63,7 @@ func TestBuildValidationFailure(t *testing.T) {
 			Phase: buildapi.BuildPhaseNew,
 		},
 	}
-	if result := ValidateBuild(build); len(result) != 3 {
+	if result := ValidateBuild(build); len(result) != 2 {
```

Comment:
it's confirming the functionality of ValidateBuild as a whole is returning multiple errors.

the specific things to be validated for are, as you stated, checked elsewhere (eg TestValidateSource)

this isn't a great test for checking validation rules, but that's not it's purpose, from what I see, so i don't see a need to change it.